File: P5/QytetetRuby/lib/calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module ModeloQytetet
class Calle < Casilla
  
attr_accessor :titulo
  
    def initialize(num,titulo_prop)
    @titulo = titulo_prop
    super(num , TipoCasilla::CALLE , @titulo.precioCompra)
  end
  
  public
  def soyEdificable()
    if(self.tipo == TipoCasilla::CALLE)
      return true
    else
      return false
    end
  end
  
  public
  def to_s()
    return super.to_s
    "Tipo: #{super.tipo}
    #{@titulo.to_s}"
    
  end
  
  def asignarPropietario(jugador)
    @titulo.propietario = jugador
  end
  
  def pagarAlquiler()
    costeAlquiler = @titulo.pagarAlquiler()
    return costeAlquiler
  end
  
  def tengoPropietario()
    return @titulo.tengoPropietario()
  end
  
  def propietarioEncarcelado()
    return @titulo.propietarioEncarcelado()
  end
end
end

File: P5/QytetetJava/src/modeloqytetet/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author romanlarrosa
 */
public class Calle extends Casilla{
    private TipoCasilla tipo;
    private TituloPropiedad titulo;
    
    public Calle(int num, TituloPropiedad _titulo){
        super(num, _titulo.getPrecioCompra());
        tipo = TipoCasilla.CALLE;
        titulo = _titulo;
    }
    
    @Override
    protected TipoCasilla getTipo(){
        return tipo;
    }
    
    @Override
    protected boolean soyEdificable(){
        if(tipo == TipoCasilla.CALLE)
           return true;
       else
           return false;
    }
    
    @Override
    public String toString(){  
        return  super.toString() +
                "\n Tipo = " + tipo + 
                "\n" + titulo.toString() + "\n";
    }
    
    @Override
    protected TituloPropiedad getTitulo(){
        return titulo;
    }
    
    public void asignarPropietario(Jugador jugador){
        titulo.setPropietario(jugador); 
    }
    
    int pagarAlquiler(){
        int costeAlquiler = titulo.pagarAlquiler();
        return costeAlquiler;
    } 
    
    private void setTitulo(TituloPropiedad _titulo){
        titulo = _titulo;
    }
    
    boolean tengoPropietario(){
        return titulo.tengoPropietario();
    }
    
    boolean propietarioEncarcelado(){
        return titulo.propietarioEncarcelado();
    }
    
    
}
